streamer/src/nonblocking/quic.rs

```diff
@@ -862,7 +864,18 @@ async fn packet_batch_sender(
                 break;
             }
 
-            let timeout_res = timeout(Duration::from_micros(250), packet_receiver.recv()).await;
```

Comment:
I've said this before: I think here we should use `coalesce - elapsed` as the timeout value and that's it?

Reply:
Can you help me understand why this if is needed?

I'd always sleep `coalesce - elapsed`. In the case a packet never comes, we wake up every `coalesce` interval, which is a high value, so effectively takes no cpu/resources. But the code becomes simpler, and we don't have to rely on the sender being dropped.

Reply:
You've marked this as resolved but the `if !packet_batch.is_empty()` is still there? I think we should always sleep `coalesce - elapsed`

Reply:
I thought about this a bit, and I mean, it's only in a test, and we're using the packet sender in the test differently from how it's used in actual code and not how it's intended (waiting for it to exit, when this is not done for server components in general in actual code). I also don't like this is dependent on the coalesce time we're given.

Reply:
> it's only in a test, and we're using the packet sender in the test differently from how it's used in actual code and not how it's intended

That's the case today, but we don't know how the code will evolve. 

>  I also don't like this is dependent on the coalesce time we're given.

Can you explain why?

With the if the logic is more complex than without the if - namely, sometimes we sleep indefinitely, sometimes we sleep on a timeout - and what is the advantage of doing so?

Reply:
> Can you explain why?
Because the coalesce time can be set to a small value, which if someone does that, would unexpectedly make the batcher spin even when there are no packets incoming.

The thing is, we're probably actually doing it wrong in the test to wait for the batcher to exit. In general we never wait for any server tasks to exit during teardown, as the infrastructure isn't there to reliably allow that (the atomic ops we use for `exit` aren't sync ops, and we would need to make fairly extensive changes to all parts of the server to make any semblance of an interface guarantee that they'll actually exit if we tell them to), and there's no point. The individual components also aren't exposed outside the server either, preventing their accidental use in that way.

I'm sympathetic to the argument to the simplicity argument, but honestly IMHO it's not that bad and just needs to be written once.

Reply:
> Because the coalesce time can be set to a small value, which if someone does that, would unexpectedly make the batcher spin even when there are no packets incoming, which isn't in the server interface or reasonably expected.

I mean, you set a timeout to a smart value, it does seem reasonable it'll spin? To me this failure mode seems better than potentially accidentally sleeping indefinitely.

> I'm sympathetic to the argument to the simplicity argument, but honestly IMHO it's not that bad and just needs to be written once.

But this argument goes both ways right? coalesce is hardcoded to 5ms, which in cpu cycles is a lifetime, so it's already been written once :P

That said if you feel strongly I won't press it, up to you.